spec/factory_spec.rb

```diff
@@ -320,6 +320,89 @@ def to_msgpack_ext
         expect(factory.dump(42)).to eq(MessagePack.dump(42))
       end
     end
+
+    describe "registering ext type with recursive serialization" do
+      before do
+        stub_const("Point", Struct.new(:x, :y, :z))
+      end
+
```

Comment:
This test doesn't check the byte order itself. If both packer&unpacker places bytes in the order of `typeid` + `length`, those tests will pass.
Could you add a test case (or assertions) to check the actual bytes from packers?

Reply:
I think the `'sets the correct length'` test handle that, by asserting that if it's parsed as an unknown ext it still works.

But I added an extra test wich directly test the bytes prefix.